lib/spack/spack/spec.py

```diff
@@ -1384,7 +1398,7 @@ def to_node_dict(self, hash_function=None, all_deps=False):
         deps = self.dependencies_dict(deptype=deptypes)
         if deps:
             if hash_function is None:
-                hash_function = lambda s: s.dag_hash()
+                hash_function = lambda s: s.dag_hash(all_deps=all_deps)
```

Comment:
I found this hard to read because all_deps is not nearly as clear as the `deptype` argument that other functions have.  I think it would be clearer that this is NOT the default case if we refactored to use `deptype=...`.